### PR TITLE
[SAJC] Add notification exit class and move general attributes to new structure 'General Information'

### PR DESCRIPTION
**Pull request by heichmann**

### CODE REVIEW

**Comment by schneidermic0** [comment]
As far as I know, we can still change the file format of SAJC incompatible. There is no use case yet (e.g., abapGit) that stores the file format outside of an ABAP server

File: file-formats/sajc/type/zif_aff_sajc_v1.intf.abap [hunk 1]
```diff
@@ -25,27 +25,39 @@ INTERFACE zif_aff_sajc_v1
...
+    BEGIN OF ty_main,
+      "! $required
+      format_version      TYPE zif_aff_types_v1=>ty_format_version,
+      "! <p class="shorttext">Header</p>
+      "! Header
+      "! $required
+      header              TYPE ty_header,
+      "! <p class="shorttext">General Information</p>
+      "! General information
+      general_information TYPE ty_general_information,
```
Comment (schneidermic0) [inline file-formats/sajc/type/zif_aff_sajc_v1.intf.abap:61]: Should this be a required field? `class_name` was required before and seems to be still required?

Comment (heichmann) [reply file-formats/sajc/type/zif_aff_sajc_v1.intf.abap:61]: Yes, this is correct. I will change it.

File: file-formats/sajc/examples/z_aff_example_sajc.sajc.json [hunk 2]
```diff
@@ -4,7 +4,10 @@
...
     "originalLanguage": "en"
   },
-  "className": "CL_APJ_SIMPLE",
-  "checkExitClass": "CL_CHECK_APJ_SIMPLE",
-  "valueHelpExitClass": "CL_VALUE_HELP_APJ_SIMPLE"
+  "generalInformation": {
+    "className": "CL_APJ_SIMPLE",
+    "checkExitClass": "CL_CHECK_APJ_SIMPLE",
+    "valueHelpExitClass": "CL_VALUE_HELP_APJ_SIMPLE",
+    "notificationExitClass": "CL_APJ_HOME_DEMO_CHECK"
```
[the remark this answers is not in the document]
Comment (heichmann) [reply file-formats/sajc/examples/z_aff_example_sajc.sajc.json:11]: O.k., we discussed it and I will change it according to your suggestion

[the remark this answers is not in the document]
Comment (heichmann) [reply file-formats/sajc/examples/z_aff_example_sajc.sajc.json]: Changes were implemented with the last commit

File: file-formats/sajc/type/zif_aff_sajc_v1.intf.abap (hunk 1 above)
Comment (heichmann) [reply file-formats/sajc/type/zif_aff_sajc_v1.intf.abap:71]: Change was implemented with the last commit

**Review by schneidermic0 — COMMENTED** [review]
Thanks for the update. I guess there was a small misunderstanding :( 

Reach out to me, if my suggestion is unclear or when you need help.

File: file-formats/sajc/type/zif_aff_sajc_v1.intf.abap [hunk 3]
```diff
@@ -24,28 +24,50 @@ INTERFACE zif_aff_sajc_v1
...
-      "! <p class="shorttext">Executed Program</p>
-      "! Name of the program which is executed within the job
-      program_name          TYPE ty_program_name,
-      "! <p class="shorttext">Class with Check Exit</p>
-      "! Name of the class which contains the check exit
-      check_exit_class      TYPE zif_aff_types_v1=>ty_object_name_30,
-      "! <p class="shorttext">Class with Value Help Exit</p>
-      "! Name of the class which contains the value help exit
-      value_help_exit_class TYPE zif_aff_types_v1=>ty_object_name_30,
+      general_information TYPE ty_general_information,
```
[the remark this answers is not in the document]
Comment (heichmann) [reply file-formats/sajc/type/zif_aff_sajc_v1.intf.abap:68]: As discussed, I moved the Exit Classes out of the General Information structure.

**Review by schneidermic0 — APPROVED** [review]
Thanks, @heichmann. Looks good to me

**Pull request merged**